ui/raidboss/timeline.js

```diff
@@ -794,22 +794,67 @@ class TimelineController {
     this.dataFiles = {};
     // data files not sent yet.
     this.timelines = null;
+
+    // Fix for #1453
+    // Countdown state
+    // 0 = No countdown
+    // 1 = Countdown active
+    // 2 = Wiped during countdown
+    this.countdownState = 0;
+    this.wipeRegex = / (?:21:........:400000(?:10|03)|00:0038:cactbot wipe)/;
   }
 
   SetPopupTextInterface(popupText) {
     this.ui.SetPopupTextInterface(popupText);
   }
 
   SetInCombat(inCombat) {
+    // On wipe, set countdown state to 2 (wiped) if it was 1 (active)
+    if (!inCombat && this.countdownState === 1)
+      this.countdownState = 2;
     if (!inCombat && this.activeTimeline)
       this.activeTimeline.Stop();
   }
 
   OnLogEvent(e) {
     if (!this.activeTimeline)
       return;
-    for (let i = 0; i < e.detail.logs.length; ++i)
+    for (let i = 0; i < e.detail.logs.length; ++i) {
+      let startMatch =
+        e.detail.logs[i].match(LocaleRegex.countdownStart[this.options.ParserLanguage]);
+      if (startMatch) {
+        // If we're not in a wipe state, reset the state to countdown active
+        if (this.countdownState !== 2)
+          this.countdownState = 1;
+      } else {
+        let engageMatch =
+          e.detail.logs[i].match(LocaleRegex.countdownEngage[this.options.ParserLanguage]);
+        if (engageMatch) {
+          let previousState = this.countdownState;
+          this.countdownState = 0;
+          // If we see an engage after a wipe, but before combat has started otherwise
+          // (e.g. countdown > wipe > face pull > engage), don't process this engage line
+          if (previousState === 2)
+            continue;
+        } else {
+          // On cancel, reset the countdown state
+          let cancelMatch =
+            e.detail.logs[i].match(LocaleRegex.countdownCancel[this.options.ParserLanguage]);
+          if (cancelMatch) {
+            this.countdownState = 0;
+          } else {
+            let wipeMatch = e.detail.logs[i].match(this.wipeRegex);
+            if (wipeMatch)
+              this.countdownState = 2;
+          }
+        }
+      }
       this.activeTimeline.OnLogLine(e.detail.logs[i]);
+      // If the previous line caused the timeline to start and we're currently in wiped state,
+      // Reset our state to account for face pull
+      if (this.activeTimeline.timebase !== null && this.countdownState === 2)
```

Comment:
This is a little awkward to check for every single log line.

It looks like engage / cancel when in state 1 sets you from state 2 to state 0 instead.  Why do you need to handle this separately?

Reply:
This is specifically to handle the case of:
1. Countdown 30s
2. Countdown 25s
3. Wipe from jumping
4. Pull boss with face pull, party decides not to wipe by jumping again

Reply:
What about doing this when "in combat" becomes true, if there's a countdown active *and* you've wiped, then clear the wipe flag.  (Maybe it's sufficient to just say "when combat becomes true, always clear the wipe flag?)

I think my main concern is doing this here on every log line vs just on the one log line of the "in combat" transition where this makes sense.

Reply:
How do I detect the line where the "in combat" transition happens given that it starts with Engage or with any timeline-specific sync to 0 e.g. `012s.txt`:
```
0.0 "Start" sync /00:0044:[^:]*:I am the Omega/ window 1,0
2.4 "--sync--" sync /:Omega-M:337D:/ window 3,0.5 # first auto
```

The only change of state to `Timeline` appears to be `timebase !== null`.

Reply:
Sorry, I meant the real "in combat" state vs the "is the timeline active" state.  I was thinking of adding more logic to `SetInCombat`.  My thought is that if combat starts (regardless of the timeline starting), then it's a real pull.

Reply:
`SetInCombat` for `TimelineController` only gets called with `false` (from `TimelineLoader.StopCombat`). 

The only way, as far as I can tell, that combat actually starts from a timeline perspective, is if a sync line matches. It never actually listens to any InCombat events or anything.

Reply:
I am suggesting to change that.  Rather than only calling `StopCombat`, instead pass `SetInCombat` all the way through.

Reply:
Alright, I agree that this probably doesn't make sense so I've dropped it in the latest version. This doesn't address one other really specific edge case though, which would be a countdown -> wipe -> abandon zone, which I believe would never fire the engage line and therefore never clear the state.